agent_communicator.go

```diff
@@ -45,12 +48,14 @@ func (a *agentCommunicator) serverHeader() string {
 	req, err := http.NewRequest(http.MethodGet, url, nil)
 
 	if err != nil {
+		a.l.Debug("Error creating request while attempting to retrieve the 'Server' header: ", err.Error())
```

Comment:
> Query: Don't we have to report it as error instead of debug?

No. This function only reports a string, expected to be the value of the header key `Server` or empty string. 
To be honest I don't expect this error to ever happen, but since http.NewRequest reports an error I took the chance to log it just in case.